src/eradiate/scenes/surface/_dem.py

```diff
@@ -83,224 +295,256 @@ class DEMSurface(Surface):
         """
-        Construct a DEM from an xarray data array holding elevation data.
-        The data array must use latitude and longitude as its dimensional
-        coordinates.
-
-        Parameters
-        ----------
-        data : DataArray
-            Data array holding the elevation data. Dimensional coordinates must
-            be latitude `lat` and longitude `lon`. Data will be interpreted to
-            be given in kernel units.
+        Mitsuba shape object identifier.
+        """
+        return f"{self.id}_shape"
 
-        bsdf : .BSDF
-            BSDF to be attached to the mesh shape.
+    @property
+    def _bsdf_id(self):
+        """
+        Mitsuba BSDF object identifier.
+        """
+        return f"{self.id}_bsdf"
 
-        planet_radius : quantity or float, optional
-            Planet radius used to convert latitude and longitude into distance
-            units. Unitless values are interpreted in default configuration
-            units. Defaults to Earth's radius.
+    def evolve_dem(self, geometry):
+        """
+        Evolve the DEM to include the atmospheric geometry.
 
-        Returns
-        -------
-        :class:`.DEMSurface`
+        In case of a spherical shell geometry, the DEM will follow the curvature of the sphere.
         """
-        radius = ensure_units(planet_radius, ucc.get("length"), convert=True).magnitude
-
-        len_lat = len(data.lat)  # x
-        len_lon = len(data.lon)  # y
-
-        # Convert the degrees to local distances based on planet radius
-        # TODO: Update this when adding support for spherical shell geometry
-        lon_rad = to_quantity(data.lon).m_as(ureg.rad)
-        lat_rad = to_quantity(data.lat).m_as(ureg.rad)
-        mean_lon = lon_rad.mean()
-        lon_range = lon_rad.max() - lon_rad.min()
-        lat_range = lat_rad.max() - lat_rad.min()
-        lon_length = lon_range * radius
-        lat_length = lat_range * radius * np.cos(mean_lon)
-
-        lat_range = list(np.linspace(-lat_length / 2.0, lat_length / 2.0, len_lat))
-        # Duplicate the first and last entry for the extra vertices, which form
-        # the vertical walls
-        lat_range.insert(0, lat_range[0])
-        lat_range.append(lat_range[-1])
-
-        lon_range = list(np.linspace(-lon_length / 2.0, lon_length / 2.0, len_lon))
-
-        vertex_positions = np.zeros(((len_lat + 2) * (len_lon + 2), 3))
-        values = np.array(data.data).transpose()[:, ::-1]
-        for i, x in enumerate(lat_range):
-            if i in [0, len_lat + 1]:
-                # For the rows of vertices which are only the lower points of the wall
-                # set the z coordinate to -1
-                z_list = np.full(len_lon + 2, -1.0)
+        theta_lim, phi_lim = _minmax_coordinates(self.shape.vertices)
+        theta_mean, phi_mean = _mean_coordinates(self.shape.vertices)
+        try:
+            atmo_bottom = geometry.ground_altitude
+        except AttributeError:
```

Comment:
When are you expecting this to happen? This attribute is defined by the `SceneGeometry` interface.